cmake/StaticAnalyzers.cmake

```diff
@@ -2,7 +2,12 @@
 # Copyright 2013-2022 The Foundry Visionmongers Ltd
 
 macro(enable_clang_tidy)
-    find_program(CLANGTIDY clang-tidy)
+    # clang-tidy-10 (default on Ubuntu 20.04, i.e. Github CI) has false
+    # positives for `performance-unnecessary-value-param` when using the
+    # idiom of immediately `std::move`ing parameters. Luckily
+    # clang-tidy-12 is available in Ubuntu 20.04 and doesn't have this
+    # problem.
+    find_program(CLANGTIDY clang-tidy-12)
```

Comment:
[question] This comment is very CI focused,  but this toolchain is generic for all builds right?

Given we do definitely need to be able to target reference platform CY2022 as well as the newer distros, is this going to cause any issues using the ASWF docker image, etc?

Reply:
It's generic in that if you enable the `OPENASSETIO_ENABLE_CLANG_TIDY` CMake option then it'll affect all builds. 

For CI we only need to enable it for one build.  So if we're (also) building in CY22 Docker, we can simply disable the clang-tidy check.

You can also override it by setting CMake variable `-DCLANGTIDY=/path/to/clang-tidy` if you really want to use a different version (I did exactly that to test different versions).

Another option is that `find_program` can take a list of possible names, in order of preference, so we could allow older versions of clang-tidy to be used as a fallback.  Obviously that means possible disagreement on the clang-tidy warnings between local and CI builds, but perhaps that's preferable over not having clang-tidy checks at all?

Reply:
I guess then, the main Q is - are we happy to say that clang tidy 12 is a base requirement for standard builds? (I know we need to enable it, but its part of our core validation suite).

The majority of real world uses of the API will be much closer to CY22 so we don't want to increase the barrier to entry of the base build checks if we can help it. I think we can't use the rational that it being available for Ubuntu 12 as a particular justification. That only helps our CI out...

If this was pylint wed just have  `# pylint: disable=performance-unnecessary-value-param` at the call site right? Is there any way we can do this here?

Reply:
Capturing off-line comment:

[fix] Phrase comment in the context of pinning on a stable `clang-tidy` version and reasons for 12 specifically. Update `BUILDING.md`

Reply:
Updated the code comment in 1eace612 and updated `CODING_STANDARDS.md` in 097d3d05 - there was no mention of `clang-tidy` in `BUILDING.md` previously, do you think we should add a note in there as well?

Reply:
> If this was pylint wed just have `# pylint: disable=performance-unnecessary-value-param` at the call site right? Is there any way we can do this here?

Indeed, `clang-tidy` supports `// NOLINT(performance-unnecessary-value-param)`-like comments.  My reason for upgrading clang-tidy rather than just suppressing the warning was that it's clearly a false positive, and is likely to crop up again and again.

Reply:
Thanks @feltech up to you re `BUILDING.md`, but I think we're good for now. Sorry I'd forgotten that we didn't mention it there.